main.tf

```diff
@@ -0,0 +1,450 @@
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "dynamodb:ExportTableToPointInTime",
+          "dynamodb:DescribeExport",
+          "dynamodb:DescribeTable",
+          "dynamodb:ListExports",
+          "dynamodb:DescribeContinuousBackups"
+        ]
+        Resource = concat(
+          # Table permissions
+          [for table_name in local.table_names :
+            "arn:aws:dynamodb:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:table/${table_name}"
+          ],
+          # Export permissions - pattern for export ARNs
+          [for table_name in local.table_names :
+            "arn:aws:dynamodb:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:table/${table_name}/export/*"
+          ]
+        )
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:PutObject",
+          "s3:PutObjectAcl",
+          "s3:GetObject",
+          "s3:ListBucket",
+          "s3:DeleteObject"
+        ]
+        Resource = [
+          data.aws_s3_bucket.mfa_backups.arn,
+          "${data.aws_s3_bucket.mfa_backups.arn}/*"
+        ]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "sts:GetCallerIdentity"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+# Daily Backup Lambda Function
+resource "aws_lambda_function" "daily_backup" {
+  filename         = data.archive_file.daily_backup.output_path
+  function_name    = "${var.app_name}-daily-backup-${var.environment}"
+  description      = "Daily ${var.app_name} Backup Lambda for ${var.environment}"
+  role             = aws_iam_role.daily_backup_lambda_role.arn
+  handler          = "lambda_function.lambda_handler"
+  runtime          = var.lambda_runtime
+  timeout          = var.lambda_timeout
+  memory_size      = var.lambda_memory_size
+  source_code_hash = data.archive_file.daily_backup.output_base64sha256
+  tags             = local.common_tags
+
+  environment {
+    variables = merge({
+      # Required environment variables
+      BACKUP_BUCKET = data.aws_s3_bucket.mfa_backups.bucket
+      ENVIRONMENT   = var.environment
+      # Table names constructed from Terraform variables
+      DYNAMODB_TABLES = jsonencode(local.table_names)
+    }, local.b2_env_vars) # Conditionally add B2 variables
+  }
+
+  depends_on = [
+    aws_iam_role_policy.daily_backup_lambda_policy,
+    aws_cloudwatch_log_group.daily_backup_logs,
+  ]
+}
+
+# CloudWatch Log Group for Daily Backup
+resource "aws_cloudwatch_log_group" "daily_backup_logs" {
+  name              = "/aws/lambda/${var.app_name}-daily-backup-${var.environment}"
+  retention_in_days = 30
+  tags              = local.common_tags
+}
+
+# EventBridge Rule for Daily Backup (using your configured schedule)
```

Comment:
The comment mentions "Daily Backup" but the schedule is open to whatever is passed in the variable. Remember to update comments added by AI to something that makes sense in context. Better yet, remove them completely because AI tends to go overboard with comments.